File: src/main/java/ninja/hudy/infosmog/update/Update.java
package ninja.hudy.infosmog.update;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface Update {
    int id() default -1;
    String description() default "";
}

File: src/main/kotlin/pl/org/smok/infosmog/update/Update.kt
package pl.org.smok.infosmog.update

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class Update(val id: Int = -1, val description: String = "")
